crates/ol/stf/src/tests_snark.rs

```diff
@@ -35,14 +180,17 @@ fn test_snark_account_deposit_and_withdrawal() {
     let snark_account_id = test_account_id(100);
     let initial_state_root = Hash::from([1u8; 32]);
 
-    // Create a NativeSnarkAccountState using the public new_fresh method
-    let snark_state = NativeSnarkAccountState::new_fresh(initial_state_root);
+    // Create a NativeSnarkAccountState with always-accept predicate key for testing
+    let vk = PredicateKey::always_accept();
+    let snark_state = NativeSnarkAccountState::new_fresh(vk, initial_state_root);
 
     let new_acct_data = NewAccountData::new_empty(AccountTypeState::Snark(snark_state));
     let snark_serial = state
         .create_new_account(snark_account_id, new_acct_data)
         .expect("Should create snark account");
 
+    // Note: Bridge gateway is a special account and doesn't need to exist in the ledger
+
     // Create a genesis block with a manifest containing a deposit to the snark account
     let deposit_amount = 150_000_000u64; // 1.5 BTC in satoshis (must be enough to cover withdrawal)
     let dest_subject = SubjectId::from([42u8; 32]);
```

Comment:
keep an eye on https://github.com/alpenlabs/alpen/pull/1229, we may need to change this a bit.